### PR TITLE
Improve compatibility with external addons

### PR DESCRIPTION
**Pull request by legokidlogan**
Allow other addons to set a pvper's base movement speed without interfering with the weapon weight system.

### CODE REVIEW

File: lua/autorun/server/sv_pvp_movespeed.lua [hunk 1]
```diff
@@ -36,19 +50,42 @@ local function movementMultiplier( totalWeight )
...
-
-    if newWalkSpeed < 100 then
-        ply:ChatPrint( "You are holding too many weapons! /drop some to regain speed." )
-        ply:SetCanWalk( false )
-    else
-        ply:SetCanWalk( true )
+    plyWraps.SetRunSpeed( ply, math.max( newRunSpeed, minRunSpeed ) )
+    plyWraps.SetWalkSpeed( ply, math.max( newWalkSpeed, minWalkSpeed ) )
+
+    local verySlow = newWalkSpeed < walkSpeedAlert
```
Comment (plally) [inline lua/autorun/server/sv_pvp_movespeed.lua:87]: walkSpeedAlert is necesarilly going to be slower than slow walking speed

Comment (legokidlogan) [reply lua/autorun/server/sv_pvp_movespeed.lua:87]: This is checking if the newly-calculated speed is slower than the alert threshold

File: lua/autorun/server/sv_pvp_movespeed.lua [hunk 2]
```diff
@@ -15,11 +17,24 @@ local weaponWeights = {
...
+local plyMeta = FindMetaTable( "Player" )
+pvpMoveSpeed.wrappedFuncs = {
+    Player = {
+        SetRunSpeed = plyMeta.SetRunSpeed,
+        SetWalkSpeed = plyMeta.SetWalkSpeed,
+    }
+}
+
+local plyWraps = pvpMoveSpeed.wrappedFuncs.Player
+local slowWalkSpeed = 100 -- Default speed when holding +walk
```
Comment (plally) [inline lua/autorun/server/sv_pvp_movespeed.lua]: Nit:

Could define this up with the other default gmod speeds

**Review by plally — DISMISSED** [review]
If you could just address that small nitpick its good.

File: lua/autorun/server/sv_pvp_movespeed.lua (hunk 2 above)
Comment (legokidlogan) [reply lua/autorun/server/sv_pvp_movespeed.lua:35]: It's not up there since it's not a config thing.

**Review by plally — APPROVED** [review]

**Pull request merged**